pkg/upgrade/upgrade_test.go

```diff
@@ -1,152 +1,251 @@
+						OSMajor:        fmt.Sprintf("%s%s", system, systemMajor),
+						OSMajorMinor:   fmt.Sprintf("%s%s.%s", system, systemMajor, systemMinor),
+						ClusterVersion: clusterVersion,
+						DriverToolkit: registry.DriverToolkitEntry{
+							ImageURL:            dtkImageURL,
+							KernelFullVersion:   kernel,
+							RTKernelFullVersion: kernelRT,
+							OSVersion:           fmt.Sprintf("%s.%s", systemMajor, systemMinor),
 						},
 					},
-				),
+				},
+			),
+		}
+
+		DescribeTable("returns information for", func(input testInput, testExpects map[string]NodeVersion) {
+			for _, labels := range input.nodesLabels {
+				node := unstructured.Unstructured{}
+				node.SetLabels(labels)
+				cache.Node.List.Items = append(cache.Node.List.Items, node)
+				cache.Node.Count = int64(len(cache.Node.List.Items))
 			}
-			table.DescribeTable("test table", func(input map[string]upgrade.NodeVersion, expected map[string]upgrade.NodeVersion) {
-				info, err := upgrade.UpdateInfo(input, defaultDTKInput, defaultImageURLInput)
-				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				gomega.Expect(info).Should(gomega.Equal(expected))
-			}, testCases...)
-		})
 
-		ginkgo.When("versions dont match", func() {
-			testCases := []table.TableEntry{
-				table.Entry(
-					"Regular kernel mismatch",
-					map[string]upgrade.NodeVersion{
-						defaultKernelFullVersionArch: upgrade.NodeVersion{
-							OSVersion:      "8.1",
-							ClusterVersion: defaultClusterVersion,
-						},
-					},
-					"OSVersion mismatch NFD: 8.1 vs. DTK: 8.4",
-				),
-				table.Entry(
-					"RT kernel mismatch",
-					map[string]upgrade.NodeVersion{
-						defaultRTKernelFullVersionArch: upgrade.NodeVersion{
-							OSVersion:      "8.2",
-							ClusterVersion: defaultClusterVersion,
-						},
-					},
-					"OSVersion mismatch NFD: 8.2 vs. DTK: 8.4",
-				),
+			mockCluster.EXPECT().VersionHistory().Return(input.clusterReleaseImages, nil)
+			mockRegistry.EXPECT().LastLayer(input.clusterReleaseImages[0]).Return(&fakeLayer{}, nil)
+			mockRegistry.EXPECT().ReleaseManifests(gomock.Any()).Return(input.clusterVersion, input.dtkImage, nil)
+			mockRegistry.EXPECT().LastLayer(input.dtkImage).Return(&fakeLayer{}, nil)
+			mockRegistry.EXPECT().ExtractToolkitRelease(gomock.Any()).Return(input.dtk, nil)
+
+			m, err := clusterInfo.GetClusterInfo()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			for expectedKernel, expectedNodeVersion := range testExpects {
+				Expect(m).To(HaveKeyWithValue(expectedKernel, expectedNodeVersion))
 			}
-			table.DescribeTable("test table", func(input map[string]upgrade.NodeVersion, expected string) {
-				_, err := upgrade.UpdateInfo(input, defaultDTKInput, defaultImageURLInput)
-				gomega.Expect(err).To(gomega.HaveOccurred())
-				gomega.Expect(err.Error()).To(gomega.Equal(expected))
-			}, testCases...)
+		}, testCases...)
+	})
+
+	When("NFD is not installed", func() {
+		It("will hint that with an error message", func() {
```

Comment:
How about removing the `When` here?
```suggestion
	It("will hint that with an error message when NFD is not installed", func() {
```

Reply:
Done